test/libgit2.jl

```diff
@@ -1906,100 +1898,88 @@ mktempdir() do dir
                 "Username for 'github.com':" => "\x04",
             ]
 
-            err, auth_attempts = challenge_prompt(ssh_empty_ex, challenges)
+            # An empty string username_ptr
+            ex = gen_ex(username="")
+            err, auth_attempts = challenge_prompt(ex, challenges)
             @test err == abort_prompt  # TODO: `eauth_error` when we can disable prompting
             @test auth_attempts == 2
 
             # A null username_ptr passed into `git_cred_ssh_key_from_agent` can cause a
             # segfault.
-            err, auth_attempts = challenge_prompt(ssh_null_ex, challenges)
+            ex = gen_ex(username=nothing)
+            err, auth_attempts = challenge_prompt(ex, challenges)
             @test err == abort_prompt  # TODO: `eauth_error` when we can disable prompting
             @test auth_attempts == 1
         end
 
         @testset "SSH explicit credentials" begin
             url = "git@github.com:test/package.jl"
+            username = "git"
 
-            invalid_key = joinpath(KEY_DIR, "invalid")
             valid_p_key = joinpath(KEY_DIR, "valid-passphrase")
-            username = "git"
             passphrase = "secret"
+            valid_cred = LibGit2.SSHCredentials(username, passphrase, valid_p_key, valid_p_key * ".pub")
 
-            valid_ex = quote
-                include($LIBGIT2_HELPER_PATH)
-                valid_cred = LibGit2.SSHCredentials($username, $passphrase, $valid_p_key, $(valid_p_key * ".pub"))
-                payload = CredentialPayload(valid_cred)
-                credential_loop(valid_cred, $url, $username, payload)
-            end
+            invalid_key = joinpath(KEY_DIR, "invalid")
+            invalid_cred = LibGit2.SSHCredentials(username, "", invalid_key, invalid_key * ".pub")
 
-            invalid_ex = quote
-                include($LIBGIT2_HELPER_PATH)
-                valid_cred = LibGit2.SSHCredentials($username, $passphrase, $valid_p_key, $(valid_p_key * ".pub"))
-                invalid_cred = LibGit2.SSHCredentials($username, "", $invalid_key, $(invalid_key * ".pub"))
-                payload = CredentialPayload(invalid_cred)
-                credential_loop(valid_cred, $url, $username, payload, use_ssh_agent=false)
+            function gen_ex(cred)
+                quote
+                    include($LIBGIT2_HELPER_PATH)
+                    payload = CredentialPayload($cred)
+                    credential_loop($valid_cred, $url, $username, payload, use_ssh_agent=false)
+                end
             end
 
             # Explicitly provided credential is correct
-            err, auth_attempts = challenge_prompt(valid_ex, [])
+            ex = gen_ex(valid_cred)
+            err, auth_attempts = challenge_prompt(ex, [])
             @test err == git_ok
             @test auth_attempts == 1
 
             # Explicitly provided credential is incorrect
-            # TODO: Unless the SSH agent is disabled we may get caught in an infinite loop
```

Comment:
Is this no longer an issue?

Reply:
Was fixed in #23641 when `use_ssh_agent=false` was set. I discovered the outdated comment while updating these tests:
https://github.com/JuliaLang/julia/blob/c172e9271d57cf2641def7fc362af2f1e71b456d/test/libgit2.jl#L1971

Reply:
Ahh, okay, thanks.